pkg/collector/corechecks/net/network/network.go

```diff
@@ -221,63 +427,275 @@ func submitProtocolMetrics(sender sender.Sender, protocolStats net.ProtoCounters
+			if len(fields) > 5 {
+				val, ok := tcpStateMetricsSuffixMapping_netstat[fields[5]]
+				if ok {
+					state := val
+					recvQ := parseQueue(fields[1])
+					sendQ := parseQueue(fields[2])
+					queueMetrics[state] = append(queueMetrics[state], recvQ, sendQ)
+				}
+			}
+		}
+	}
+
+	return queueMetrics, nil
+}
+
+func parseQueue(queueStr string) uint64 {
+	var queue uint64
+	_, err := fmt.Sscanf(queueStr, "%d", &queue)
+	if err != nil {
+		return 0
+	}
+	return queue
+}
+
+func submitConnectionsMetrics(sender sender.Sender, protocolName string, stateMetricSuffixMapping map[string]string, connectionsStats []net.ConnectionStat, collectConnectionQueues bool, ssAvailable bool) {
 	metricCount := map[string]float64{}
 	for _, suffix := range stateMetricSuffixMapping {
 		metricCount[suffix] = 0
 	}
-
 	for _, connectionStats := range connectionsStats {
 		metricCount[stateMetricSuffixMapping[connectionStats.Status]]++
 	}
-
 	for suffix, count := range metricCount {
 		sender.Gauge(fmt.Sprintf("system.net.%s.%s", protocolName, suffix), count, "", nil)
 	}
+	queueMetrics := make(map[string][]uint64)
+	if collectConnectionQueues && protocolName[:3] == "tcp" {
+		var queues map[string][]uint64
+		var err error
+		// pass in version number
+		if ssAvailable {
+			queues, err = getQueueMetrics(protocolName[len(protocolName)-1:])
+			if err != nil {
+				log.Debug("Error getting queue metrics with ss:", err)
+				return
+			}
+		} else {
+			queues, err = getQueueMetricsNetstat(protocolName[len(protocolName)-1:])
+			if err != nil {
+				log.Debug("Error getting queue metrics with netstat:", err)
+				return
+			}
+		}
+		for state, queues := range queues {
+			queueMetrics[state] = append(queueMetrics[state], queues...)
+		}
+		for state, queues := range queueMetrics {
+			for _, queue := range queues {
+				sender.Histogram("system.net.tcp.recv_q", float64(queue), "", []string{"state:" + state})
+				sender.Histogram("system.net.tcp.send_q", float64(queue), "", []string{"state:" + state})
+			}
+		}
+	}
 }
 
-func netstatTCPExtCounters() (map[string]int64, error) {
-	f, err := os.Open("/proc/net/netstat")
-	if err != nil {
-		return nil, err
+func netstatAndSnmpCounters(procfsPath string, protocolNames []string) (map[string]net.ProtoCountersStat, error) {
+	counters := make(map[string]net.ProtoCountersStat)
+	for _, protocol := range protocolNames {
+		counters[protocol] = net.ProtoCountersStat{Protocol: protocol, Stats: make(map[string]int64)}
 	}
-	defer f.Close()
 
-	counters := map[string]int64{}
-	scanner := bufio.NewScanner(f)
-	for scanner.Scan() {
-		line := scanner.Text()
-		i := strings.IndexRune(line, ':')
-		if i == -1 {
-			return nil, errors.New("/proc/net/netstat is not fomatted correctly, expected ':'")
-		}
-		proto := strings.ToLower(line[:i])
-		if proto != "tcpext" {
-			continue
+	for _, subdirectory := range []string{"netstat", "snmp"} {
+		fs := filesystem
+		f, err := fs.Open(procfsPath + "/net/" + subdirectory)
```

Comment:
We seem to be concatenating  the same string `procfsPath + "/net/" + subdirectory` multiple times. Should store the value once and use it multiple times? 

Also, we might benefit from using `filepath.Join(procfsPath, 'net', subdirectory)` instead of concatenating manually  

https://pkg.go.dev/path/filepath#Join

Reply:
fixed in https://github.com/DataDog/datadog-agent/pull/33190/commits/ba29632a0a1849681c4d535a283c13679680fc68